### ISSUE TITLE
Spurious test failure in t/filter.t proves the module works

### ISSUE THREAD

**Issue by nanis** [perlancar/perl-lib-filter#4]
This is on Windows 8.1 Pro with 64-bit `perl` 5.22.0 compiled using Visual Studio 2013.

<pre>C:\Users\...\.cpanm\work\1435273844.1008\lib-filter-0.10> prove -b t\filter.t
t\filter.t .. Module 'List::Util' is disallowed (disallow) at C:\Users\...\.cpanm\work\1435273844.1008\lib-filter-0.10\blib\lib/lib/filter.pm line 103.
Compilation failed in require at C:/opt/perl-5.22.0/site/lib/Socket.pm line 869.
Compilation failed in require at C:/opt/perl-5.22.0/lib/IO/Socket.pm line 13.
BEGIN failed--compilation aborted at C:/opt/perl-5.22.0/lib/IO/Socket.pm line 13.
Compilation failed in require at -e line 1.
BEGIN failed--compilation aborted at -e line 1.

    #   Failed test 'req IO::Socket ok'
    #   at C:/Users/.../.cpanm/work/1435273844.1008/lib-filter-0.10/t/lib/testlib.pl line 28.
    # child_err=65280
...</pre>

[`Socket.pm` line 869](https://metacpan.org/source/PEVANS/Socket-2.020/Socket.pm#L863):

```
if( defined &getaddrinfo ) {
    # These are not part of the API, nothing uses them, and deleting them
    # reduces the size of %Socket:: by about 12K
    delete $Socket::{fake_getaddrinfo};
    delete $Socket::{fake_getnameinfo};
} else {
    require Scalar::Util; ### line 869

    *getaddrinfo = \&fake_getaddrinfo;
    *getnameinfo = \&fake_getnameinfo;
```

In [Scalar::Util](https://metacpan.org/source/PEVANS/Scalar-List-Utils-1.42/lib/Scalar/Util.pm#L22), we have:

```
require List::Util; # List::Util loads the XS
```

However, `List::Util` is disallowed by the test:

```
test_lib_filter(
    name => 'disallow',
    args => ["disallow", "IPC::Cmd;List::Util"],
    require_nok => ["IPC::Cmd", "List::Util"],
    require_ok  => ["IO::Socket"], # core
);
```

Therefore, it is impossible to satisfy the `require_ok => ["IO::Socket"]` portion of the test.

As I mentioned before, it might be more reliable to test functionality by bundling dummy modules in the test directory.

**Comment by nanis** [comment]
I should point out that in all my years of using Perl on Windows, I had not realized `$Config{d_getaddrinfo}` was undefined. I am not sure why that is the case. See also [Why is $Config{d_getaddrinfo} undef on Windows?](https://www.reddit.com/r/perl/comments/3b4t1q/why_is_configd_getaddrinfo_undef_on_windows/) on /r/perl.

Not directly relevant, but just pointing out that I would not have anticipated this test failing due to `perl` not using the platform's `getaddrinfo` on Windows.

If anything, this spurious test failure is actual confirmation that `lib::filter` works ;-)

**Comment by perlancar** [comment]
Nice to know about the side effect of using a real module in the test :)

I've just changed almost all the the modules in the test to dummy ones. Previously I had just refactored the tests and had not changed the module names.

Released 0.11.
